components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorService.java

```diff
@@ -80,8 +80,9 @@ public JsonObject getAuthenticationConfigurations(String appName)
         grantTypes.add(KeyManagerConstants.PASSWORD_GRANT_TYPE);
         grantTypes.add(KeyManagerConstants.AUTHORIZATION_CODE_GRANT_TYPE);
         grantTypes.add(KeyManagerConstants.REFRESH_GRANT_TYPE);
-        APIMStoreConfigurations storeConfigs = ServiceReferenceHolder.getInstance().getAPIMStoreConfiguration();
-        String callBackURL = storeConfigs.getApimBaseUrl() + "login/callback/" + appName;
+        APIMAppConfigurations appConfigs = ServiceReferenceHolder.getInstance()
```

Comment:
Check whether we can get to a single line as 
ServiceReferenceHolder.getInstance()..getAPIMAppConfiguration();